salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +53,103 @@ def node_info(node, ca_minion):
     return result
 
 
+def get_cluster_version(api_client):
+    coreV1 = kubernetes.client.CoreV1Api(api_client=api_client)
+    try:
+        namespace = coreV1.read_namespace(name="kube-system")
+    except (ApiException, HTTPError) as exc:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to read namespace information {}'.format(exc)
+        )
+    annotation_key = 'metalk8s.scality.com/cluster-version'
+
+    if annotation_key not in namespace.metadata.annotations:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return namespace.metadata.annotations[annotation_key]
+
+
+def iso_timestamp_converter(timestamp):
+    if not timestamp:
+        return None
+    return timestamp.isoformat()
+
+
+def get_storage_classes(api_client):
+    storageV1 = kubernetes.client.StorageV1Api(api_client=api_client)
+    storage_classes = {}
+    try:
+        storageclass_list = storageV1.list_storage_class()
+    except (ApiException, HTTPError) as exc:
+        return __utils__['pillar_utils.errors_to_dict'](
```

Comment:
should be `return {}`, because at the call site you have this code

```python
    storage_classes = get_storage_classes(api_client)

    if not storage_classes:
        return __utils__['pillar_utils.errors_to_dict'](
            'Cannot list storage classes'
        )
```

Reply:
I think it's a good idea to get the content of the exception

Reply:
Fine by me, but then you need to update the calling code otherwise it's broken…

Or does `__utils__` stuff returns None already?

Reply:
Well, we can actually return None and log the exception accordingly

Reply:
`__utils__|'pillarutils.errors_to_dict']` return a dict with `_errors` key with a list of messages